### PR TITLE
[DESK-703] feature logs device

### PR DESCRIPTION
**Pull request by mauricio0000**
### Description
[DESK-703] feature logs device

### Checklist

- [ ] Pull Request title is in the format `[PROJ-12345] Some useful description here...`
- [ ] I have prefixed this PR with `[WIP]` if it is a Work In Progress (not ready to merge)
- [ ] The git branch is in the format `PROJ-12345-some-useful-description-here`
- [ ] I have added one or more reviewers
- [ ] I have added/updated tests for any code changes
- [ ] I have updated documentation (including README, inline comments and docs.remote.it docs)
- [ ] This PR only contains one isolated change (bug fix, feature, etc)
- [ ] Translation strings added/updated for all user-visible text
- [ ] I have manually reviewed this PR to make sure only the files I intended to change were changed and nothing else
- [ ] I have notified reviewers on Slack by @ mentioning them in the `#desktop`channel
- [ ] I have moved the Jira ticket into `Resolved` state and made a note in the ticket with a link to this PR

### Test plan

1. Sign in
2. Go to Device
3. Click Device Details
4. Click Logs
4. You should  list of logs

### Ticket

https://remoteit.atlassian.net/browse/DESK-703

[DESK-703]: https://remoteit.atlassian.net/browse/DESK-703

### CODE REVIEW

File: backend/src/sharedCopy/applications.ts
Comment (JamieRuderman) [inline backend/src/sharedCopy/applications.ts:113]: This is great - looks like it's handling international date formatting, but I don't think it belongs in the applications.ts file. How about a new helper file? `/helpers/dateHelper.ts` maybe?

File: common/types.d.ts
Comment (JamieRuderman) [inline common/types.d.ts:286]: This doesn't look like it's being used?

File: frontend/src/buttons/CSVDownloadButton/CSVDownloadButton.tsx
Comment (JamieRuderman) [inline frontend/src/buttons/CSVDownloadButton/CSVDownloadButton.tsx:32]: Would you mind adding a `<Tooltip title="Download CSV">` wrapper here please?

File: frontend/src/services/graphQLDevice.ts
Comment (JamieRuderman) [inline frontend/src/services/graphQLDevice.ts:70]: Let's wait till we can merge in the Account linking branch, then break out the log query into its own model and request only on page load. We can use the new single device api.

File: frontend/src/services/graphQLDevice.ts
Comment (JamieRuderman) [inline frontend/src/services/graphQLDevice.ts:170]: We will want the single device query here as it's much more performant. But again let's wait for account linking as it's changed how to query a bit.

File: package.json
Comment (JamieRuderman) [inline package.json:34]: I don't believe these are needed in the root level package.json for electron, are they?

**Review by JamieRuderman — COMMENTED** [review]
Looking great. Just a few questions. Also let's just hold off on merging until we can merge account linking to future and resolve conflicts and optimize the queries a little. 

Thanks @mauricio0000 !

File: frontend/src/components/DatePicker/DatePicker.tsx
Comment (JamieRuderman) [inline frontend/src/components/DatePicker/DatePicker.tsx:10]:
```suggestion
  onChange: (date: any) => void
```

File: frontend/src/models/devices.ts
Comment (JamieRuderman) [inline frontend/src/models/devices.ts:149]: I think we should be getting logs independent from device list. Please reference the accounts.ts or applicationTypes.ts models and let's make a logs.ts model to handle the querying and data.

File: frontend/src/services/graphQLDevice.ts
Comment (JamieRuderman) [inline frontend/src/services/graphQLDevice.ts:67]: Let's not just get these logs when we get the device list, instead break these queries out into their own graphQLLogs.ts file and let's get them on demand when we go to the page. That will be much better for device list load times.

File: frontend/src/services/graphQLDevice.ts
Comment (JamieRuderman) [reply frontend/src/services/graphQLDevice.ts:199]: You should now be able to reference the single device query from account linking.

File: frontend/src/pages/DeviceLogPage/DeviceLogPage.tsx
Comment (JamieRuderman) [inline frontend/src/pages/DeviceLogPage/DeviceLogPage.tsx:82]:
```suggestion
              onChange={handleChange}
```

**Review by JamieRuderman — CHANGES_REQUESTED** [review]
Thanks @mauricio0000 ! Just some architectural changes and update for working with account linking. Thanks!

File: frontend/src/buttons/CSVDownloadButton/CSVDownloadButton.tsx
Comment (JamieRuderman) [inline frontend/src/buttons/CSVDownloadButton/CSVDownloadButton.tsx:24]: Just curious why you do this as a saved state and effect? Is that to prevent multiple clicks?

File: frontend/src/services/graphQLDevice.ts
Comment (JamieRuderman) [inline frontend/src/services/graphQLDevice.ts:92]: Looks like you are getting the logs in it's own model now, can we remove this from the device list query?

**Review by JamieRuderman — CHANGES_REQUESTED** [review]
Looking great just would love to get the logs query out of the device list query, then done! 
Thank you!

File: frontend/src/buttons/CSVDownloadButton/CSVDownloadButton.tsx
Comment (mauricio0000) [reply frontend/src/buttons/CSVDownloadButton/CSVDownloadButton.tsx:24]: we need to wait for the new eventURL after click in download button

**Review by JamieRuderman — APPROVED** [review]
Great! Approved.

**Pull request merged**
